### PR TITLE
[uss_qualifier] Add GetAccessTokens scenario

### PR DESCRIPTION
**Pull request by BenjaminPelletier**
This PR defines, documents, and implements a new GetAccessTokens scenario to address #1629 using new InterUSS requirements that a test designer can use to express/capture authorization-related requirements of their interoperability ecosystem.  A new dev test configuration that exercises this scenario is added and invoked from the CI.

### CODE REVIEW

File: monitoring/uss_qualifier/resources/communications/auth_adapter.py [hunk 1]
```diff
@@ -25,10 +26,14 @@ class AuthAdapterSpecification(ImplicitDict):
     scopes_authorized: list[str]
     """List of scopes the user in the auth spec is authorized to obtain."""
 
+    participant_id: Optional[ParticipantID]
+    """Identify of participant responsible for providing the authorization source used by this auth adapter."""
```
Comment (barroco) [inline monitoring/uss_qualifier/resources/communications/auth_adapter.py]:
```suggestion
    """Identify participant responsible for providing the authorization source used by this auth adapter."""
```

File: monitoring/uss_qualifier/configurations/dev/access_tokens.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                    MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAtjrMt+vOxuqjOU+hwrVA
+                    gHjBMs9nMw1ONSpiLOpUQSnqvBwB0Zba+8e2tGJeQDWEf8KoVt3PfMa34EomLhWZ
+                    GWEosftV8gZSbneDjUnE+kUTXvZmMFa2Byc9njsl5N7dOC1pT7qHNJOB9RhK/9eh
+                    Pl7XczRuIJR7N/ZFk+XZSlFh5werq7ME7GdhBPFaO30KZRmRhgwVtAP2kA8sSrqg
+                    xXuuyy808UvT3MkzL63Sv/EzQSs8YLVAn/BwiQlmWxKFmtzNk9RJdYkxRJN1E9E2
+                    sS2i6b55j94hiPwwQ16GsAAWUT+ijezlHuGZU+OLYtYfoHy+4Ku31doQc0ujvtYO
+                    JQIDAQAB
+                    -----END PUBLIC KEY-----
+                  expires_in_more_than: 50m
+                  expires_in_less_than: 1h10m
```
Comment (the-glu) [inline monitoring/uss_qualifier/configurations/dev/access_tokens.yaml]: It's a bit special, but have you considered moving it to `has_claim_values` to have all claims at the same place?

File: monitoring/uss_qualifier/scenarios/interuss/communications/get_access_tokens.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+                        try:
+                            public_pem = fix_key(
+                                expect.expectations.validates_against_public_key
+                            )
+                            # Perform signature verification while intentionally bypassing audience/timestamp checks
+                            # to isolate signature evaluation and prevent false positives from expired/immature tokens.
+                            jwt.decode(
+                                token,
+                                key=public_pem,
+                                algorithms=[header.get("alg", "RS256")],
```
Comment (the-glu) [inline monitoring/uss_qualifier/scenarios/interuss/communications/get_access_tokens.py:115]: nit/minor: Getting the algorithm from the header itself is not correct, it should come from the expected algorithm (based on the key or expectations.alg), to avoid some attacks.

**Review by the-glu — COMMENTED** [review]
The scenario only allows for positive results (e.g., "With these settings, I get this token").

Shouldn't we also test the negative case? For example: "With this unknown scope, I don't get a token because it shouldn't be allowed."

This may be out of scope for these specific tests, but it could be valuable for validating authorization servers.

File: monitoring/uss_qualifier/scenarios/interuss/communications/get_access_tokens.py (hunk 3 above)
Comment (BenjaminPelletier) [reply monitoring/uss_qualifier/scenarios/interuss/communications/get_access_tokens.py:115]: I agree that when validating an access token for the purpose of granting access, the algorithm should definitely be supplied rather than taken from the header so a fake token couldn't be generated with a weak algorithm.  In this case though, I would expect a test designer to specify an expected `alg` to achieve the same effective result.  I'll add additional warning documentation regarding this expectation.

File: monitoring/uss_qualifier/configurations/dev/access_tokens.yaml (hunk 2 above)
Comment (BenjaminPelletier) [reply monitoring/uss_qualifier/configurations/dev/access_tokens.yaml]: I'm not sure I follow -- there are expectations that could be folded into has_xyz_values (e.g., `alg` into `has_header_values`), but I don't think this is one unless we introduce ASTExpressions.  The value in `exp` is an absolute time, so to achieve the effect of these expires_in_xyz fields, we would need a `greater_than_expr` expression that is something like `"($request_time + parse_time_delta('50m')).timestamp()"` and specifically populate the `request_time` variable and `parse_time_delta` function when evaluating that expression.  It seems clearer and more straightforward to do it this way.

**Review by BenjaminPelletier — COMMENTED** [review]
> Shouldn't we also test the negative case? For example: "With this unknown scope, I don't get a token because it shouldn't be allowed."

This could be added as an additional feature but I think the content of this PR contains a self-contained feature (this change isn't needed to make the PR complete), this PR is already pretty big, and #1629 doesn't ask for this capability.  So, I think it's probably out of scope for this PR at least.

File: monitoring/uss_qualifier/configurations/dev/access_tokens.yaml (hunk 2 above)
Comment (the-glu) [reply monitoring/uss_qualifier/configurations/dev/access_tokens.yaml]: The idea was to move in into the 'has_claim_values' object, but with a similar name and do the request_time + X computation there as well, without any other complexity, just the same two posibilities.

This would also have the advantage of testing 'nbf' and 'iat' (and possibility others claims with date) with similar checks.

Comment (BenjaminPelletier) [reply monitoring/uss_qualifier/configurations/dev/access_tokens.yaml]: Ah, good point regarding `nbf` and `iat` -- I'll make the suggested adjustment.

**Review by the-glu — APPROVED** [review]
👍

File: monitoring/uss_qualifier/resources/communications/auth_adapter.py (hunk 1 above)
Comment (barroco) [inline monitoring/uss_qualifier/resources/communications/auth_adapter.py]:
```suggestion
    """Identifier of participant responsible for providing the authorization source used by this auth adapter."""
```

**Review by barroco — APPROVED** [review]
